src/Page/Simulator/Impact.elm

```diff
@@ -0,0 +1,68 @@
+module Page.Simulator.Impact exposing (..)
+
+import Data.Unit as Unit
+import Html exposing (..)
+import Html.Attributes as Attr exposing (..)
+import Html.Events exposing (..)
+
+
+type alias Config msg =
+    { selected : Impact
+    , switch : Result String Impact -> msg
+    }
+
+
+type Impact
+    = ClimateChange
+    | FreshwaterEutrophication
+
+
+impacts : List Impact
+impacts =
+    [ ClimateChange
+    , FreshwaterEutrophication
+    ]
+
+
+impactToString : Impact -> String
+impactToString impact =
+    case impact of
+        ClimateChange ->
+            "Changement climatique"
+
+        FreshwaterEutrophication ->
+            "Eutrophisation d'eaux douces"
+
+
+impactFromString : String -> Result String Impact
+impactFromString string =
+    case string of
+        "Changement climatique" ->
+            Ok ClimateChange
+
+        "Eutrophisation d'eaux douces" ->
+            Ok FreshwaterEutrophication
+
+        _ ->
+            Err <| "Impact inconnu: " ++ string
```

Comment:
The thing I like with returning an error instead of falling back on a "default" value is that I can render an error message to end users in case an external string value couldn't be mapped to something we know. This is especially useful for decoders, but it makes sense in a DOM context as well, where browser extensions can sometimes mess around with form input values for example. So being a robust and informative strategy, I'd really like to keep it that way.